### PR TITLE
Report if key is not set

### PR DESCRIPTION
**Pull request by shubham7109**
## Description
Display dialog if an API key is not set prompting the user to set an API key.

## Links and Data
Issue: [Link](https://devtopia.esri.com/runtime/kotlin/issues/5019)

### CODE REVIEW

File: app/src/main/java/com/esri/arcgismaps/kotlin/sampleviewer/ui/screens/home/HomeCategoryScreen.kt [hunk 1]
```diff
@@ -116,10 +119,25 @@ fun HomeCategoryScreen(
                     CategoryCard(category) { onNavigateToCategory(category.title) }
                 }
             }
+            CheckAccessToken()
         }
     }
 }
 
+@Composable
```
Comment (TADraeseke) [inline app/src/main/java/com/esri/arcgismaps/kotlin/sampleviewer/ui/screens/home/HomeCategoryScreen.kt:127]: Maybe just a function level comment here to remind us what this is doing in a few years time :D

File: app/src/main/java/com/esri/arcgismaps/kotlin/sampleviewer/ui/screens/home/HomeCategoryScreen.kt [hunk 2]
```diff
@@ -116,10 +119,25 @@ fun HomeCategoryScreen(
...
+fun CheckAccessToken() {
+    var isDialogDisplayed by remember {
+        mutableStateOf(BuildConfig.ACCESS_TOKEN == "DEFAULT_ACCESS_TOKEN")
+    }
+    if (isDialogDisplayed) {
+        MessageDialog(
+            title = "Using default access token",
```
Comment (TADraeseke) [inline app/src/main/java/com/esri/arcgismaps/kotlin/sampleviewer/ui/screens/home/HomeCategoryScreen.kt:134]:
```suggestion
            title = "Access token",
```

Might look better on one line

File: app/src/main/java/com/esri/arcgismaps/kotlin/sampleviewer/ui/screens/home/HomeCategoryScreen.kt [hunk 3]
```diff
@@ -116,10 +119,25 @@ fun HomeCategoryScreen(
...
+            description = "Add an ACCESS_TOKEN to the project's local.properties",
```
Comment (TADraeseke) [inline app/src/main/java/com/esri/arcgismaps/kotlin/sampleviewer/ui/screens/home/HomeCategoryScreen.kt:135]: Do we want to include a hyperlink to the guide doc here?

**Review by TADraeseke — CHANGES_REQUESTED** [review]
Nice fix!

See below for suggestions.

Also, we should extract these strings to strings.xml

File: app/src/main/java/com/esri/arcgismaps/kotlin/sampleviewer/ui/screens/home/HomeCategoryScreen.kt (hunk 3 above)
Comment (shubham7109) [reply app/src/main/java/com/esri/arcgismaps/kotlin/sampleviewer/ui/screens/home/HomeCategoryScreen.kt:143]: I cannot add a link to the string params of the `MessageDialog` instead, I added a link to repo instructions on setting the key as part of the function comment.

File: app/src/main/java/com/esri/arcgismaps/kotlin/sampleviewer/ui/screens/home/HomeCategoryScreen.kt [hunk 4]
```diff
@@ -116,10 +119,33 @@ fun HomeCategoryScreen(
                     CategoryCard(category) { onNavigateToCategory(category.title) }
                 }
             }
+            CheckAccessToken()
         }
     }
 }
 
+/**
+ * Display a dialog if the project does not have an access token set.
```
Comment (TADraeseke) [inline app/src/main/java/com/esri/arcgismaps/kotlin/sampleviewer/ui/screens/home/HomeCategoryScreen.kt:128]:
```suggestion
 * Display a dialog if the default access token placeholder is still set.
```

**Review by TADraeseke — APPROVED** [review]
LGTM! Just slight clarification that this won't catch expired or otherwise incorrect API keys

**Pull request merged**